src/render/drm_format_set.zig

```diff
@@ -0,0 +1,33 @@
+const FlexibleArrayType = @import("std").zig.c_translation.FlexibleArrayType;
+
+pub const DrmFormat = extern struct {
+    format: u32 align(8),
+    len: usize,
+    capacity: usize,
+    pub fn modifiers(self: anytype) FlexibleArrayType(@TypeOf(self), u64) {
+        const Intermediate = FlexibleArrayType(@TypeOf(self), u8);
+        const ReturnType = FlexibleArrayType(@TypeOf(self), u64);
+        return @ptrCast(ReturnType, @alignCast(@alignOf(u64), @ptrCast(Intermediate, self) + 24));
```

Comment:
Hardcoding 24 here isn't correct for e.g. 32 bit targets, note that the output of `zig translate-c` is target specific in some cases like this. `zig translate-c` gets this constant from libclang [here](https://github.com/ziglang/zig/blob/0298442100b2d5707099f09d29af554e8c6ac87e/src/translate_c.zig#L972).

I wonder if we shouldn't just import the function generated by translate-c directly instead. I haven't looked into duplicating clang's struct layout logic, I'm assuming that would be non-trivial.

Reply:
Hey, thanks! That makes sense. I hadn't really looked at what the `FlexibleArrayType` stuff was doing and just copied it blindly.

The alignment of the `format` field will also change depending on the CPU arch, so I've implemented a way to get that. Let me know if there's a better way to get it as I'm still fairly new to Zig.

Reply:
> The alignment of the `format` field will also change depending on the CPU arch, so I've implemented a way to get that. Let me know if there's a better way to get it as I'm still fairly new to Zig.

Good point, In practice I think this should always be `8` because of how the C ABI lays out structs and because there are not currently any targets with 128 bit pointers but matching the translate-c output here is better.

Reply:
The alignment of the `format` field is `4` when running translate-c on a `i386`.

I only know the basics of struct alignment. But I assume this is something to do with the fact that the following fields are `usize` and also 32-bits on a 32-bit platform, so there isn't a need for padding out the alignment of the `format` field before them.

Reply:
Interesting, I think that's a bug. The alignment should be 8 to ensure that the `u64`s in the flexible array member are properly aligned.

Reply:
Opened https://github.com/ziglang/zig/issues/10944 to track this.

Reply:
To test this some more, I've just compiled the following C code on Intel x64 and i386:
```c
#include <stdio.h>
#include <stddef.h>
#include <stdint.h>
#include <stdalign.h>

struct foo1 {
    uint32_t format;
    size_t len;
    size_t capacity;
    
    uint64_t modifiers[];
};

int main(int argc, char *argv)
{
    printf("sizeof(uint32_t)                  = %zu\n", sizeof(uint32_t));
    printf("sizeof(size_t)                    = %zu\n", sizeof(size_t));
    printf("sizeof(uint64_t)                  = %zu\n", sizeof(uint64_t));
    printf("sizeof(struct foo1)               = %zu\n", sizeof(struct foo1));

    printf("offsetof(struct foo1, modifiers)  = %zu\n", offsetof(struct foo1, modifiers));
    printf("alignof(struct foo1)              = %zu\n", alignof(struct foo1));
}
```

On x64, I get:
```
sizeof(uint32_t)                  = 4
sizeof(size_t)                    = 8
sizeof(uint64_t)                  = 8
sizeof(struct foo1)               = 24
offsetof(struct foo1, modifiers)  = 24
alignof(struct foo1)              = 8
```

And on i386, I get:
```
sizeof(uint32_t)                  = 4
sizeof(size_t)                    = 4
sizeof(uint64_t)                  = 8
sizeof(struct foo1)               = 12
offsetof(struct foo1, modifiers)  = 12
alignof(struct foo1)              = 4
```

So, at first glance it seems Zig is doing the right thing here?

Reply:
Aha, thanks for doing that testing. I've learned something today, namely that the alignment of `uint64_t` (aka `unsigned long long` on `i386`) has alignment `4` on i386. See the ABI spec here: https://uclibc.org/docs/psABI-i386.pdf

Sorry for the confusion, I'll close that misguided Zig bug report and merge this!